x-pack/plugins/enterprise_search/server/lib/ml_inference_pipeline/get_inference_errors.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  AggregationsMultiBucketAggregateBase,
+  AggregationsStringRareTermsBucketKeys,
+} from '@elastic/elasticsearch/lib/api/types';
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+export interface MlInferenceError {
+  message: string;
+  doc_count: number;
+  timestamp: string | undefined; // Date string
+}
+
+export interface ErrorAggregationBucket extends AggregationsStringRareTermsBucketKeys {
+  max_error_timestamp: {
+    value: number | null;
+    value_as_string?: string;
+  };
+}
+
+/**
+ * Fetches an aggregate of distinct ML inference errors from the target index, along with the most
+ * recent error's timestamp and affected document count for each bucket.
+ * @param indexName the index to get the errors from.
+ * @param esClient the Elasticsearch Client to use to fetch the errors.
+ */
+export const getMlInferenceErrors = async (
+  indexName: string,
+  esClient: ElasticsearchClient
+): Promise<MlInferenceError[]> => {
+  const searchResult = await esClient.search<
+    unknown,
+    {
+      errors: AggregationsMultiBucketAggregateBase<ErrorAggregationBucket>;
```

Comment:
Why is there a semicolon at the end of this line? 🤔

Reply:
It is a type declaration, so it follows the same pattern similar to the line 21